internal/cli/create/create.go

```diff
@@ -1,62 +1,126 @@
 package create
 
 import (
-	"strings"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
 
-	"github.com/pkg/errors"
+	"connectrpc.com/connect"
+	pkgerrors "github.com/pkg/errors"
 	"github.com/spf13/cobra"
-	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/cli-runtime/pkg/printers"
 	"k8s.io/utils/pointer"
+	sigyaml "sigs.k8s.io/yaml"
 
 	"github.com/akuity/kargo/internal/cli/client"
 	"github.com/akuity/kargo/internal/cli/option"
+	kargosvcapi "github.com/akuity/kargo/pkg/api/service/v1alpha1"
 )
 
+type Flags struct {
+	Filenames []string
+}
+
 func NewCommand(opt *option.Option) *cobra.Command {
+	var flag Flags
 	cmd := &cobra.Command{
-		Use:   "create (RESOURCE) (NAME)",
-		Short: "Create a resource",
-		Args:  cobra.MinimumNArgs(2),
+		Use:   "create [--project=project] -f (FILENAME)",
+		Short: "Create a resource from a file or from stdin",
 		Example: `
+# Create a stage using the data in stage.yaml
+kargo create -f stage.yaml
+
 # Create project
 kargo create project my-project
 `,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			ctx := cmd.Context()
-			kargoSvcCli, err := client.GetClientFromConfig(ctx, opt)
-			if err != nil {
-				return errors.New("get client from config")
+			var namespace *string
+			if opt.Project.IsPresent() {
+				v, _ := opt.Project.Get()
+				namespace = &v
+			}
+			if len(flag.Filenames) == 0 {
+				return pkgerrors.New("filename is required")
 			}
 
-			resource := strings.ToLower(strings.TrimSpace(args[0]))
-			if resource == "" {
-				return errors.New("resource is required")
+			currentDir, err := os.Getwd()
+			if err != nil {
+				return pkgerrors.Wrap(err, "get current directory")
 			}
 
-			var res runtime.Object
-			switch resource {
-			case "project", "projects":
-				name := args[1]
-				res, err = createProject(ctx, kargoSvcCli, name)
+			var printer printers.ResourcePrinter
+			if pointer.StringDeref(opt.PrintFlags.OutputFormat, "") != "" {
+				printer, err = opt.PrintFlags.ToPrinter()
 				if err != nil {
-					return errors.Wrap(err, "create project")
+					return pkgerrors.Wrap(err, "new printer")
 				}
-			default:
-				return errors.Errorf("unknown resource %q", resource)
 			}
 
-			if pointer.StringDeref(opt.PrintFlags.OutputFormat, "") == "" {
-				_ = printers.NewTablePrinter(printers.PrintOptions{}).PrintObj(res, opt.IOStreams.Out)
-				return nil
-			}
-			printer, err := opt.PrintFlags.ToPrinter()
+			kargoSvcCli, err := client.GetClientFromConfig(ctx, opt)
 			if err != nil {
-				return errors.Wrap(err, "new printer")
+				return pkgerrors.Wrap(err, "get client from config")
 			}
-			return printer.PrintObj(res, opt.IOStreams.Out)
+
+			var successRes []*kargosvcapi.CreateResourceResult_CreatedResourceManifest
+			var createErrs []error
+			for _, filename := range flag.Filenames {
```

Comment:
I think looping through files and sending them one at a time might be problematic...

See [this comment](https://github.com/akuity/kargo/pull/592#discussion_r1304943410) on #592.

I think concern about proper order to create resources forces you to concatenate all the manifest files together and submit them as one.